### PR TITLE
chore: update claude+Copilot instructions based on YAML config migration progress [ci skip]

### PR DESCRIPTION
**Pull request by shouples**
Many changes have been applied to the server config and underlying logic over the past couple of weeks, and our current "AI guidance" has drifted, so this PR is catching up to some of the most recent changes by removing outdated references like `getRequiredEnvVars` in favor of the new config and related expectations.

Some QoL extras:
- new `mcp-docs` skill to help look up [MCP documentation](https://modelcontextprotocol.io/docs)
- new `tool-handlers` rule for finer guidance that doesn't need to go in `CLAUDE.md`

### CODE REVIEW

File: .claude/rules/unit-tests.md [hunk 1]
```diff
@@ -139,14 +139,24 @@ object. **Do not reach for `vi.mock`** - if a new dependency seems to require it
...
-- Test config (`getToolConfig()`), required env vars (`getRequiredEnvVars()`), and behavior (`handle()`)
-- Stub `ClientManager` methods with `createMockInstance(DefaultClientManager)`
+- Test `getToolConfig()` (name, description, input schema, annotations).
+- Test `enabledConnectionIds(runtime)` against representative `ServerRuntime` fixtures: a runtime
+  whose connection has the relevant service block (expect the connection id) and a bare runtime
+  without that block (expect `[]`). Helper factories live in `tests/factories/runtime.ts`
+  (`flinkRuntime()`, `tableflowRuntime()`, `bareRuntime()`, etc.).
+- Test `handle()` for typical and edge-case inputs.
+- Stub `ClientManager` methods with `createMockInstance(DefaultClientManager)`.
 - Use `as any` only on partial mock return values (e.g., a mock admin client with only
```
Comment (shouples) [inline .claude/rules/unit-tests.md:149]: `as any` is prime for future tech debt cleanup, but not in this branch

File: .claude/rules/unit-tests.md [hunk 2]
```diff
@@ -139,14 +139,24 @@ object. **Do not reach for `vi.mock`** - if a new dependency seems to require it
...
+
+## Test Server & Runtime Fixtures
+
+- `createTestServer(clientManager, toolNames?)` from `@tests/server.js` boots a real `McpServer`
+  with all tools (or a passed subset) wired to a stubbed `ClientManager` over `InMemoryTransport`.
+  Use it for protocol-level integration tests.
+- For `enabledConnectionIds()` tests, build a `ServerRuntime` via the named factories in
+  `@tests/factories/runtime.js`: `bareRuntime()`, `kafkaRuntime()`, `flinkRuntime()`,
+  `tableflowRuntime()`, `schemaRegistryRuntime()`, `confluentCloudRuntime()`,
+  `telemetryRuntime()`, etc. For uncommon shapes, use `runtimeWith(env, connectionConfig)`.
```
[the remark this answers is not in the document]
Comment (shouples) [reply .claude/rules/unit-tests.md]: https://github.com/confluentinc/mcp-confluent/pull/281/commits/b49b107f22263efbd56fef278bfa265e2e7b7228

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -8,8 +8,8 @@ assistants. The tool surface splits into two groups:
 - **Kafka-protocol tools** that work against any Apache Kafka®-compatible cluster or Schema
   Registry (e.g., topic CRUD, producing and consuming messages, schema management).
 - **Confluent Cloud-specific tools** that wrap CCloud REST APIs (e.g., Flink, Tableflow,
-  billing), automatically disabled when no `confluent_cloud` block is present in the connection
-  config (YAML) or when the corresponding environment variables are not set.
+  billing), enabled only when the relevant service block (`confluent_cloud:`, `flink:`,
+  `tableflow:`) is present in a connection's YAML config.
```
[the remark this answers is not in the document]
Comment (shouples) [reply .github/copilot-instructions.md]: https://github.com/confluentinc/mcp-confluent/pull/281/commits/657d87503a02a7d7d090bdebac4658619620f72e

File: .github/copilot-instructions.md [hunk 4]
```diff
@@ -23,24 +23,38 @@ this codebase. Author-facing guidance (how to scaffold a tool, run the inspector
 
 ### Entry flow
 
-At startup the server parses CLI arguments, loads and validates configuration, constructs the
-shared client manager, builds the set of enabled tools, registers them, and starts the configured
-transports.
+At startup: `parseCliArgs()` → `buildConfigFromEnvAndCli()` (resolves a single
+`MCPServerConfiguration` from either a YAML file via `-c <path>` or legacy env vars + CLI args)
+→ `ServerRuntime.fromConfig()` constructs one `DefaultClientManager` per connection → iterates
+`ToolName` enum to build the enabled tool set → registers tools on `McpServer` → starts transports.
```
[the remark this answers is not in the document]
Comment (shouples) [reply .github/copilot-instructions.md]: https://github.com/confluentinc/mcp-confluent/pull/281/commits/4d8c367a1040c83830f03fc0bab053560d2a6a8d

File: .github/copilot-instructions.md [hunk 5]
```diff
@@ -23,24 +23,38 @@ this codebase. Author-facing guidance (how to scaffold a tool, run the inspector
...
+during the migration, from env vars + CLI args). Each handler declares its requirement via
+`enabledConnectionIds(runtime)`, which returns the ids of connections that satisfy a predicate
+from `src/confluent/tools/connection-predicates.ts`. An empty result disables the tool.
 
 ### Key layers
 
+- **`src/config/`** — Configuration core. `models.ts` defines `MCPServerConfiguration` (a Zod
+  schema with per-service connection blocks: `kafka`, `flink`, `schema_registry`,
+  `confluent_cloud`, `tableflow`, `telemetry`). `env-config.ts` exports
+  `buildConfigFromEnvAndCli()`, the single entry point that resolves config from YAML or env vars.
```
[the remark this answers is not in the document]
Comment (shouples) [reply .github/copilot-instructions.md]: https://github.com/confluentinc/mcp-confluent/pull/281/commits/4d8c367a1040c83830f03fc0bab053560d2a6a8d

File: CLAUDE.md [hunk 6]
```diff
@@ -30,25 +30,28 @@ Pre-commit hook runs `npm run format && npm run lint` automatically via Husky.
...
 
 ### Key Layers
 
+- **`src/config/`** — Configuration core. `models.ts` defines `MCPServerConfiguration` (a Zod schema with per-service connection blocks: `kafka`, `flink`, `schema_registry`, `confluent_cloud`, `tableflow`, `telemetry`). `env-config.ts` exports `buildConfigFromEnvAndCli()` (re-exported from `index.ts`), the single entry point that produces a config from either a YAML file (`-c <path>`) or env vars + CLI args. `interpolation.ts` handles `${VAR}` interpolation inside YAML.
```
[the remark this answers is not in the document]
Comment (shouples) [reply CLAUDE.md]: https://github.com/confluentinc/mcp-confluent/pull/281/commits/a709e036a8c3dece31893be01780fff3909a3b07

File: CLAUDE.md [hunk 7]
```diff
@@ -30,25 +30,28 @@ Pre-commit hook runs `npm run format && npm run lint` automatically via Husky.
 
 ### Entry Point & Startup Flow
 
-`src/index.ts` → `parseCliArgs()` → `initEnv()` (Zod-validated env vars) → creates `DefaultClientManager` → iterates `ToolName` enum to build enabled tool set → registers tools on `McpServer` → starts transports.
+`src/index.ts` → `parseCliArgs()` → `buildConfigFromEnvAndCli()` (re-exported via `src/config/index.ts`, resolves a single `MCPServerConfiguration` from either a YAML file or env vars + CLI args, see issue #151) → `ServerRuntime.fromConfig()` constructs a `DefaultClientManager` per connection → iterates `ToolName` enum to build enabled tool set → registers tools on `McpServer` → starts transports.
```
[the remark this answers is not in the document]
Comment (shouples) [reply CLAUDE.md]: https://github.com/confluentinc/mcp-confluent/pull/281/commits/a709e036a8c3dece31893be01780fff3909a3b07

File: .claude/rules/tool-handlers.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+## Handler Structure
+
+Every tool handler follows this pattern:
+
+1. **Pick a base class:**
+   - If a domain subclass exists (e.g., `FlinkToolHandler`, `TableflowToolHandler` under `handlers/<domain>/`), extend it. The subclass already implements `enabledConnectionIds()` for the whole domain, so you don't repeat the predicate per tool.
+   - Otherwise extend `BaseToolHandler` from `@src/confluent/tools/base-tools.js` directly and implement `enabledConnectionIds()` yourself. If a domain has more than one tool with the same predicate, introduce a domain subclass first and extend that.
+2. **Implement these methods (skip `enabledConnectionIds` if you inherited it):**
+   - `getToolConfig()` → returns `{ name: ToolName, description: string, inputSchema: zodSchema.shape, annotations: ToolAnnotations }`. Use one of the shared annotation constants exported from `base-tools.ts`: `READ_ONLY`, `CREATE_UPDATE`, or `DESTRUCTIVE` (don't construct ad-hoc instances).
+   - `handle(clientManager, toolArguments, sessionId?)` → returns `Promise<CallToolResult>`.
```
Comment (jlrobins) [inline .claude/rules/tool-handlers.md:17]: Is about to change ;-(, but is truthy for next hours

File: .claude/rules/unit-tests.md [hunk 9]
```diff
@@ -139,14 +139,24 @@ object. **Do not reach for `vi.mock`** - if a new dependency seems to require it
...
+## Test Server & Runtime Fixtures
+
+- `createTestServer(clientManager, toolNames?)` from `@tests/server.js` boots a real `McpServer`
+  with all tools (or a passed subset) wired to a stubbed `ClientManager` over `InMemoryTransport`.
+  Use it for protocol-level integration tests.
+- For `enabledConnectionIds()` tests, build a `ServerRuntime` via the named factories in
+  `@tests/factories/runtime.js`: `bareRuntime()`, `kafkaRuntime()`, `flinkRuntime()`,
+  `tableflowRuntime()`, `schemaRegistryRuntime()`, `confluentCloudRuntime()`,
+  `telemetryRuntime()`, etc. For uncommon shapes, use `runtimeWith(connectionConfig?, connectionId?)` (both args optional; `connectionId` defaults to `"default"`).
+- Integration tests (`*.integration.test.ts`) default to real I/O; stub selectively. Unit tests
```
Comment (jlrobins) [inline .claude/rules/unit-tests.md]: Does not exist yet?

**Review by jlrobins — APPROVED** [review]
Small things to consider, but much better than before!

**Review by jlrobins — APPROVED** [review]

File: .claude/rules/unit-tests.md (hunk 9 above)
Comment (shouples) [reply .claude/rules/unit-tests.md]: Context bleedover, fixed in https://github.com/confluentinc/mcp-confluent/pull/281/commits/cb7189a58792e89d11aca53b3825b86f2f2af5ea - good catch!
